### DISCUSSION TITLE
Please set the environment variable 'CUDA_VISIBLE_DEVICES' correctly before trying to run GANDLF on GPU

### DISCUSSION THREAD

**Discussion by HILALOZTEMEL** [mlcommons/GaNDLF#445] [Q&A]
I'm trying to train model in cuda but ı get an error message. how can ı fix it ?

**Comment by sarthakpati** [comment]
To run GaNDLF on the GPU, you will need to set the environment variable `CUDA_VISIBLE_DEVICES` [[1](https://developer.nvidia.com/blog/cuda-pro-tip-control-gpu-visibility-cuda_visible_devices/), [2](https://cbica.github.io/GaNDLF/usage#running-gandlf-traininginference)] to the specific GPU you want to run computations on.

**Reply by sarthakpati** [reply]
That being said, if you are using AMD, you might need to do something different. Since I have no experience with AMD cards, this is perhaps something you could help contribute to GaNDLF! 😄

**Reply by HILALOZTEMEL** [reply]
I got this error on another computer . This computer uses nvidia

**Reply by HILALOZTEMEL** [reply]
i want to run cuda 

like this :

python gandlf_run  -c ./quadrantSegmentation/model.yaml -i ./quadrantSegmentation/train.csv -m ./quadrantSegmentation/model -t True -d cuda

**Reply by HILALOZTEMEL** [reply]
i get an error  when I  write this code `python gandlf_run -c ./quadrantSegmentation/model.yaml -i ./quadrantSegmentation/train.csv -m ./quadrantSegmentation/model -t True -d cuda`

**Reply by sarthakpati** [reply]
Did you set the environment variable, as I said?

**Reply by Geeks-Sid** [reply]
Also, it is very helpful if you post the error message as well in order for us to help you better.
